pkg/cmd/server/start/start_node.go

```diff
@@ -225,17 +226,17 @@ func (o NodeOptions) CreateNodeConfig() error {
 	return nil
 }
 
-func RunSDNController(config configapi.NodeConfig) {
+func RunSDNController(config configapi.NodeConfig, nodeConfig *kubernetes.NodeConfig) {
 	if config.NetworkPluginName == osdn.NetworkPluginName() {
-		kclient, _, err := configapi.GetKubeClient(config.MasterKubeConfig)
-		if err != nil {
-			glog.Fatal("Failed to get kube client for SDN")
-		}
 		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
 		if err != nil {
 			glog.Fatal("Failed to get kube client for SDN")
 		}
-		osdn.Node(*oclient, *kclient, config.NodeName, "")
+		ch := make(chan struct{})
+		nodeConfig.KubeletConfig.StartUpdates = ch
+		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
```

Comment:
They're different?

> On Jun 9, 2015, at 12:01 AM, Jordan Liggitt <notifications@github.com> wrote:
> 
> In pkg/cmd/server/start/start_node.go:
> 
> >  		oclient, _, err := configapi.GetOpenShiftClient(config.MasterKubeConfig)
> >  		if err != nil {
> >  			glog.Fatal("Failed to get kube client for SDN")
> >  		}
> > -		osdn.Node(*oclient, *kclient, config.NodeName, "")
> > +		ch := make(chan struct{})
> > +		nodeConfig.KubeletConfig.StartUpdates = ch
> > +		go osdn.Node(oclient, nodeConfig.Client, config.NodeName, "", ch)
> why use the node config's client instead of getting our own?
> 
> —
> Reply to this email directly or view it on GitHub.
>